engine/include/Engine/Intermediate/GameObject.inl

```diff
@@ -0,0 +1,179 @@
+{
+    return m_name;
+}
+
+void GameObject::setName(const char* newName) noexcept
+{
+    m_name = newName;
+}
+
+constexpr const TransformComponent& GameObject::getTransform() const noexcept
+{
+    return *m_pTransform;
+}
+
+constexpr TransformComponent& GameObject::getTransform() noexcept
+{
+    return *m_pTransform;
+}
+
+template <typename T, typename... Args>
+GameObject& GameObject::addChild(Args&&... args) noexcept
+{
+    std::unique_ptr<GameObject>& pChild = this->children.emplace_back(std::make_unique<T>(args...));
+    pChild->children                    = std::list<std::unique_ptr<GameObject>>();
+    // pChild->update((*this).getModelMatrix());
+    pChild->parent = this;
+    return *pChild;
+}
+
+template <typename T>
+std::vector<T*> GameObject::getComponents() noexcept
+{
+    std::vector<T*> toReturn;
+    for (auto&& pComponent : m_pComponents)
+    {
+        T* comp = dynamic_cast<T*>(pComponent);
+
+        if (comp != nullptr)
+        {
+            toReturn.emplace_back(comp);
+        }
+    }
+    return toReturn;
+}
+
+constexpr std::list<Component*>& GameObject::getComponents() noexcept
+{
+    return m_pComponents;
+}
+
+constexpr const std::list<Component*>& GameObject::getComponents() const noexcept
+{
+    return m_pComponents;
+}
+
+void GameObject::setTag(const std::string& newTag) noexcept
+{
+    m_tag = newTag;
+}
+
+constexpr const std::string& GameObject::getTag() const noexcept
+{
+    return m_tag;
+}
+
+std::list<std::unique_ptr<GameObject>>::iterator GameObject::destroyChild(
+    const std::list<std::unique_ptr<GameObject>>::iterator& it) noexcept
+{
+    return children.erase(it);
+}
+
+template <typename TUniqueComponentType>
+void GameObject::destroyImmediateUniqueComponent() noexcept
+{
+    for (auto&& it = m_pComponents.begin(); it != m_pComponents.end(); ++it)
+    {
+        TUniqueComponentType* checkedCompPtr = dynamic_cast<TUniqueComponentType*>(*it);
+
+        if (checkedCompPtr != nullptr)
+            [[unlikely]]
+            {
+                DataChunk<TUniqueComponentType>::getInstance()->destroyComponent(checkedCompPtr);
+                m_pComponents.erase(it);
+                return;
+            }
+    };
+}
+
+void GameObject::setActive(bool newState)
+{
+    for (auto&& i : m_pComponents)
+    {
+        i->setActive(newState);
+    }
+}
+
+std::list<Component*>::iterator GameObject::destroyComponent(const std::list<Component*>::iterator& it) noexcept
+{
+    return m_pComponents.erase(it);
+}
+
+void GameObject::destroy() noexcept
+{
+    /*set flag to be delete by it parent*/
+    m_isDead = true;
+    // m_isDirty = true;
+}
+
+void GameObject::destroyImmediate() noexcept
+{
+    parent->destroyChild(this);
+}
+
+constexpr bool GameObject::operator==(GameObject const& other) noexcept
+{
+    return (this == &other);
+}
+
+bool GameObject::compareTag(const std::string& toCompare) const noexcept
+{
+    if (toCompare.compare(m_tag) == 0)
```

Comment:
Just return (toCompare.compare(m_tag) == 0) already, there is no need for a branch.

**Idea**
You could also return without comparison.
Explanation : 
0 is false
other is true
so if you return (!toCompare.compare(m_tag)), it would return the correct result.